llvm-project/Dockerfile.epoch4

```diff
@@ -0,0 +1,300 @@
+    llvm-profdata merge --output=${arch}.profdata.prof \
+      $(find ${LLVM_BUILD_DIR} -name \*.profraw) && \
+    find ${LLVM_BUILD_DIR} -name \*.profraw | xargs rm
+
+ARG arch=hexagon
+RUN ARCH=${arch} HOSTCFLAGS=${SYSROOT} HOSTLDFLAGS=${SYSROOT} \
+      make -C linux-5.18-rc6 LLVM=1 -s -j$(nproc) allnoconfig all && \
+    llvm-profdata merge --output=${arch}.profdata.prof \
+      $(find ${LLVM_BUILD_DIR} -name \*.profraw) && \
+    find ${LLVM_BUILD_DIR} -name \*.profraw | xargs rm
+
+# Mips needs bash if building all targets.
+ARG arch=mips
+RUN ARCH=${arch} HOSTCFLAGS=${SYSROOT} HOSTLDFLAGS=${SYSROOT} \
+      make -C linux-5.18-rc6 LLVM=1 -s -j$(nproc) allnoconfig vmlinux && \
+    llvm-profdata merge --output=${arch}.profdata.prof \
+      $(find ${LLVM_BUILD_DIR} -name \*.profraw) && \
+    find ${LLVM_BUILD_DIR} -name \*.profraw | xargs rm
+
+# powernv_defconfig needs find -printf to work. find in busybox in alpine
+# doesn't support that flag.
+ARG arch=powerpc
+RUN ARCH=${arch} HOSTCFLAGS=${SYSROOT} HOSTLDFLAGS=${SYSROOT} \
+      make -C linux-5.18-rc6 LLVM=1 -s -j$(nproc) allnoconfig vmlinux && \
+    llvm-profdata merge --output=${arch}.profdata.prof \
+      $(find ${LLVM_BUILD_DIR} -name \*.profraw) && \
+    find ${LLVM_BUILD_DIR} -name \*.profraw | xargs rm
+
+# riscv requires perl to build
+RUN apk add perl findutils
+ARG arch=riscv
+RUN ARCH=${arch} HOSTCFLAGS=${SYSROOT} HOSTLDFLAGS=${SYSROOT} \
+      make -C linux-5.18-rc6 LLVM=1 -s -j$(nproc) defconfig vmlinux && \
+    llvm-profdata merge --output=${arch}.profdata.prof \
+      $(find ${LLVM_BUILD_DIR} -name \*.profraw) && \
+    find ${LLVM_BUILD_DIR} -name \*.profraw | xargs rm
+RUN apk del perl findutils
+
+# s390 needs CC=clang, but Alpine does not package GNU binutils for s390 in
+# main, so we need to fetch it from testing.
+RUN apk add binutils-s390x --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing/
+# We did not build openssl-dev or elfutils-dev, so hack them into the sysroot.
+# Don't reuse the sysroot from stage4!
+# s390 needs bash and GNU find.
+RUN apk add openssl openssl-dev python3 elfutils-dev bash findutils
+RUN \
+  ln -s /usr/include/gelf.h /sysroot/usr/include/. && \
+  ln -s /usr/include/libelf.h /sysroot/usr/include/. && \
+  ln -s /usr/lib/libelf.so /sysroot/usr/lib/. && \
+  ln -s /usr/include/openssl/ /sysroot/usr/include/. && \
+  ln -s /usr/lib/libcrypto.so /sysroot/usr/lib/.
+# bpf/resolve_btfids doesn't respect HOSTCFLAGS
+RUN cp -r /sysroot/usr/include/ /usr/local/
+# TODO: something in the s390 build isn't respecting HOSTCC.
+RUN \
+  ln -s $(which clang) /usr/local/bin/gcc && \
+  ln -s $(which clang) /usr/local/bin/s390x-alpine-linux-musl-gcc
+
+ARG arch=s390
+#ARCH=s390 CROSS_COMPILE=s390x-linux-gnu- make CC=clang -j72 defconfig
+RUN ARCH=${arch} HOSTCFLAGS=${SYSROOT} HOSTLDFLAGS=${SYSROOT} \
+      CROSS_COMPILE=s390x-alpine-linux-musl- \
+      CC=clang LLVM_IAS=0 \
```

Comment:
As we mentioned offline, if we upgrade to a 5.19 prerelease, we can drop the `LLVM_IAS=0` here.